File: check_korean.js
const fs = require('fs');

function checkNonAscii() {
    try {
        const content = fs.readFileSync('src/contexts/LanguageContext.tsx', 'utf-8');
        const lines = content.split('\n');
        
        let enStart = -1;
        let krStart = -1;
        
        for (let i = 0; i < lines.length; i++) {
            if (lines[i].includes('EN: {')) enStart = i;
            if (lines[i].includes('KR: {')) { krStart = i; break; }
        }
        
        if (enStart === -1 || krStart === -1) {
            console.log('Could not find EN or KR blocks');
            return;
        }
        
        const matches = [];
        
        for (let i = enStart; i < krStart; i++) {
            const line = lines[i].trim();
            if (line === '') continue;
            // Check for Korean characters OR the ₩ symbol
            if (/[가-힣]/.test(line) || line.includes('₩')) {
                matches.push({ lineNum: i + 1, content: line });
            }
        }
        
        console.log(`Found ${matches.length} lines with Korean or Won characters in EN block.`);
        for (let i = 0; i < matches.length; i++) {
            console.log(`Line ${matches[i].lineNum}: ${matches[i].content}`);
        }
        
    } catch (err) {
        console.error('Error:', err);
    }
}

checkNonAscii();

File: .safe_gate/legacy_archive/check_korean.py
import re

def check_korean():
    try:
        with open('src/contexts/LanguageContext.tsx', 'r', encoding='utf-8') as f:
            lines = f.readlines()
            
        en_start = -1
        kr_start = -1
        
        for i, line in enumerate(lines):
            if 'EN: {' in line:
                en_start = i
            if 'KR: {' in line:
                kr_start = i
                break
                
        if en_start == -1 or kr_start == -1:
            print("Could not find EN or KR blocks")
            return
            
        korean_pattern = re.compile(r'[가-힣]')
        matches = []
        for i in range(en_start, kr_start):
            line = lines[i]
            if korean_pattern.search(line):
                matches.append((i+1, line.strip()))
                
        print(f"Found {len(matches)} lines with Korean characters in EN block.")
        for i, (line_num, line) in enumerate(matches[:20]):
            print(f"Line {line_num}: {line}")
            
    except Exception as e:
        print(f"Error: {e}")

if __name__ == '__main__':
    check_korean()
